lib_bgp_data/blacklist/blacklist_parser.py

```diff
@@ -49,62 +50,77 @@ def _run(self):
-        output_str = dict()
-        for typ in sources.keys():
-            output_str[typ] = ''
-        # For each source in the sources dict, GET url and save string
+
+
+        # TODO: If UCEPROTECT blank, retry DONE
+        # Download sources to file, and then read from file instead of reading from string, makes more sense/readable DONE
+        # Why am I using an IP for UCE's mirrors? The various mirrors of UCE's blacklists are not consistent with one another:
+        # Some will return a gzip, some just ISO-8859 text, and some just don't work. So, to ensure consistency, I'm using IP here.
+        # This should return plaintext.
+        sources = {'uce2': 'http://72.13.86.154/rbldnsd-all/dnsbl-2.uceprotect.net.gz', 'uce3': 'http://72.13.86.154/rbldnsd-all/dnsbl-3.uceprotect.net.gz', 'spamhaus': 'https://www.spamhaus.org/drop/asndrop.txt', 'mit': 'https://raw.githubusercontent.com/ctestart/BGP-SerialHijackers/master/prediction_set_with_class.csv'}
+        output_path = dict()
+        # For each source in the sources dict, GET url, write response to path, and save path
         # to dict.
-        for source in sources.keys():
+        for source in sources:
             downloaded_file = requests.get(sources[source])
-            # UCEPROTECT is inconsistent in that sometimes a python
-            # requested file will sometimes come as a txt, sometimes
-            # as a tarball that needs to be unzipped. We check this:
-            if downloaded_file.headers['Content-Type'] == 'application/octet-stream':
-                decompressed = gzip.decompress(downloaded_file.content)
-                # UCEPROTECT uses ISO-8859-1 for text, not utf-8
-                output_str[source] = decompressed.decode('ISO-8859-1')
-            else:
-                # We have a txt and save to dict directly
-                output_str[source] = downloaded_file.text
-        return output_str
+            # Now we'll check if we got a proper file, if not try 5 more times or until success
+            if not downloaded_file.ok or downloaded_file.content == '':
+                for i in range(0, 5):
+                     # Don't want to spam the page
+                     time.sleep(5)
+                     downloaded_file = requests.get(sources[source])
+                     if downloaded_file.ok and downloaded_file.content != '':
+                         break
+            if not downloaded_file.ok or downloaded_file.content == '':
+                print("Aborting: File from " + source + 
+                      " failed to download properly with status code" + 
+                      str(downloaded_file.status_code) + "and length" + 
+                      str(len(downloaded_file.content)))
+                sys.exit()
```

Comment:
don't exit like this, it will exit with exit code 0 (no error). But even so, we shouldn't sys exit, and rather should raise an exception is the more pythonic way